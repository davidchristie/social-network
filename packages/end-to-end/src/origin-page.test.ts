import puppeteer, { Browser, Page } from "puppeteer";


describe("origin page", () => {
  let browser: Browser;
  let page: Page;

  beforeAll(async () => {
    browser = await puppeteer.launch();
    page = await browser.newPage();
    await page.goto("http://localhost:3000");
  });

  afterAll(async () => {
    await browser.close();
  });

  it(`redirects to "/login"`, async () => {
    await page.waitForNavigation();
    expect(await page.url()).toEqual("http://localhost:3000/login");
  });

  it("renders LoginPage component", async () => {
    expect(await page.$(".LoginPage")).not.toBeNull();
  });

  it("matches snapshot", async () => {
    expect(await page.content()).toMatchSnapshot();
  });
});
