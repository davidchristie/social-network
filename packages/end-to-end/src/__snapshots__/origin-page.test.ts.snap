// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`origin page matches snapshot 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">
    <meta name=\\"theme-color\\" content=\\"#000000\\">
    <link rel=\\"manifest\\" href=\\"/manifest.json\\">
    <link rel=\\"shortcut icon\\" href=\\"/favicon.ico\\">
    <title>Social Network</title>
  <style type=\\"text/css\\">.Alert {
  background-color: red;
  color: white;
  font-size: 1.4em;
  margin-bottom: 1em;
  padding: 1em;
}
</style><style type=\\"text/css\\">.Button {
  -ms-flex-align: center;
      align-items: center;
  background-color: white;
  border: solid black 1px;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1.1em;
  opacity: 0.6;
  -webkit-transition: all 200ms ease-in-out;
  -o-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.Button:hover {
  opacity: 1;
  -webkit-transform: scale(1.02);
      -ms-transform: scale(1.02);
          transform: scale(1.02);
}
</style><style type=\\"text/css\\">.ButtonLink {
  text-decoration: none;
}
</style><style type=\\"text/css\\">.Header {
  -ms-flex-align: center;
      align-items: center;
  border: 2px solid black;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: justify;
      justify-content: space-between;
  padding: 1em;
}
</style><style type=\\"text/css\\">.Layout {
  height: 100vh;
}
</style><style type=\\"text/css\\">.Section {
  border: solid black 2px;
  padding: 1em;
}

@media only screen and (min-width: 600px) {
  .Section {
    padding: 2em;
  }
}
</style><style type=\\"text/css\\">.Input {
  font-size: 1.4em;
  width: 100%;
}
</style><style type=\\"text/css\\">.ProfilePosts .Post {
  border: 1px solid black;
  margin-top: 1em;
  padding: 2em;
}
</style><style type=\\"text/css\\">.Application {
  font-family: Arial, sans-serif;
  font-size: 1.4em;
}

.Application a {
  color: inherit;
}
</style><style type=\\"text/css\\">body {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  margin: 0;
}

div {
  margin: 0;
}
</style></head>
  <body>
    <noscript>
      You need to enable JavaScript to run this app.
    </noscript>
    <div id=\\"root\\"><div class=\\"Application\\"><div class=\\"Layout\\"><header class=\\"Header\\"><div><a href=\\"/\\">Social Network</a></div><div><a class=\\"ButtonLink\\" href=\\"/signup\\"><button class=\\"Button\\">Signup</button></a><a class=\\"ButtonLink\\" href=\\"/login\\"><button class=\\"Button\\">Login</button></a></div></header><div class=\\"LoginPage\\"><section class=\\"Section\\"><form><h2>Login</h2>Don't have an account? <a href=\\"/signup\\">Click here to sign up.</a><hr><div><label for=\\"access-account-email\\">Email</label><input class=\\"Input\\" id=\\"access-account-email\\" name=\\"email\\" required=\\"\\" type=\\"email\\" value=\\"\\"></div><div><label for=\\"access-account-password\\">Password</label><input class=\\"Input\\" id=\\"access-account-password\\" name=\\"password\\" required=\\"\\" type=\\"password\\" value=\\"\\"></div><button class=\\"Button\\">Submit</button></form></section></div></div></div></div>
  <script type=\\"text/javascript\\" src=\\"/static/js/bundle.js\\"></script>

</body></html>"
`;
